cmd/kubeadm/app/cmd/alpha/certs.go

```diff
@@ -68,6 +74,47 @@ var (
 
 	You can also use "kubeadm init --upload-certs" without specifying a certificate key and it will
 	generate and print one for you.
+`)
+	generateCSRLongDesc = cmdutil.LongDesc(`
+	Generates keys and certificate signing requests (CSRs) for all the certificates required to run the control plane.
+	This command also generates partial kubeconfig files with private key data in the  "users > user > client-key-data" field,
+	and for each kubeconfig file an accompanying ".csr" file is created.
+
+	This command is designed for use in [Kubeadm External CA Mode](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-certs/#external-ca-mode).
+	It generates CSRs which you can then submit to your external certificate authority for signing.
+
+	The PEM encoded signed certificates should then be saved alongside the key files, using ".crt" as the file extension,
+	or in the case of kubeconfig files, the PEM encoded signed certificate should be base64 encoded
+	and added to the kubeconfig file in the "users > user > client-certificate-data" field.
+`)
+	generateCSRExample = cmdutil.Examples(`
+	# The following command will generate keys and CSRs for all control-plane certificates and kubeconfig files:
+	kubeadm alpha certs generate-csr --kubeconfig-dir /tmp/etc-k8s --cert-dir /tmp/etc-k8s/pki
+	# ...and it produces the following directories and files:
+	tree /tmp/etc-k8s
+	/tmp/etc-k8s
+	├── admin.conf
+	├── admin.conf.csr
+	├── controller-manager.conf
+	├── controller-manager.conf.csr
+	├── pki
+	│   ├── apiserver.csr
+	│   ├── apiserver-etcd-client.csr
+	│   ├── apiserver-etcd-client.key
+	│   ├── apiserver.key
+	│   ├── apiserver-kubelet-client.csr
+	│   ├── apiserver-kubelet-client.key
+	│   ├── etcd
+	│   │   ├── healthcheck-client.csr
+	│   │   ├── healthcheck-client.key
+	│   │   ├── peer.csr
+	│   │   ├── peer.key
+	│   │   ├── server.csr
+	│   │   └── server.key
+	│   ├── front-proxy-client.csr
+	│   └── front-proxy-client.key
+	├── scheduler.conf
+	└── scheduler.conf.csr
```

Comment:
i'd prefer if we don't hardcode the list list of CSR/keys as part of the example, because if we end up changing something this hardcoded list will drift.

IMO the first part is suffcient:
```
# The following command will generate keys and CSRs for all control-plane certificates and kubeconfig files:
kubeadm alpha certs generate-csr --kubeconfig-dir /tmp/etc-k8s --cert-dir /tmp/etc-k8s/pki
```